src/transaction/script.c

```diff
@@ -0,0 +1,257 @@
+#include <stdbool.h>  // bool
+#include <string.h>   // memset, explicit_bzero, memmove
+
+#include "os.h"  // PRINTF
+
+#include "script.h"
+#include "../constants.h"
+#include "opcodes.h"
+
+/**
+ * Identifies if the given script is P2PKH
+ */
+bool identify_p2pkh_script(buffer_t *in, uint16_t script_len) {
+    uint8_t p2pkh[] = {OP_DUP, OP_HASH160, PUBKEY_HASH_LEN, OP_EQUALVERIFY, OP_CHECKSIG};
+    size_t offset = 0;
+
+    if ((in == NULL) || ((script_len != 25) && (script_len != 31)) ||
+        (in->size - in->offset < script_len)) {
+        PRINTF("[-] [Identify P2PKH script] invalid args (script_len=%d, size=%d, offset=%d)\n",
+               script_len,
+               in->size,
+               in->offset);
+        return false;
+    }
```

Comment:
We check for NULL pointers, for the expected length since p2pkh can only be 25 or 31 bytes long (without and with timelock) and lastly if the input buffer has enough data and capacity to read the script.